proxy/envoy/resources.go

```diff
@@ -18,96 +18,67 @@ import (
-	// shared by all proxy instances. Since Istio does not assign a local
-	// service/service version to each proxy instance, the name is same for all
-	// of them. This setting corresponds to "--service-cluster" flag in Envoy.
-	// The value for "--service-node"  is used by the proxy to identify its set
-	// of local instances to RDS for source-based routing. For example, if proxy
-	// sends its IP address, the RDS can compute routes that are relative to the
-	// service instances located at that IP address.
-	IstioServiceCluster string
+	// DefaultLbType defines the default load balancer policy
+	DefaultLbType = LbTypeRoundRobin
 
-	// DiscoveryRefreshDelay is the average delay Envoy uses between
-	// fetches to the SDS/CDS/RDS APIs.
-	DiscoveryRefreshDelay time.Duration
-}
-
-var (
-	// DefaultMeshConfig configuration
-	DefaultMeshConfig = &MeshConfig{
-		DiscoveryAddress:          "manager:8080",
-		MixerAddress:              "mixer:9091",
-		ProxyPort:                 15001,
-		AdminPort:                 15000,
-		BinaryPath:                "/usr/local/bin/envoy",
-		ConfigPath:                "/etc/envoy",
-		EnableAuth:                false,
-		AuthConfigPath:            "/etc/certs",
-		DrainTimeSeconds:          30,
-		ParentShutdownTimeSeconds: 45,
-		IstioServiceCluster:       "istio-proxy",
-		DiscoveryRefreshDelay:     1000 * time.Millisecond,
-	}
-)
+	// RDSName is the name of RDS cluster
+	RDSName = "rds"
 
-// TODO: these values used in the Envoy configuration will be configurable
-const (
-	DefaultTimeoutMs = 1000
-	DefaultLbType    = LbTypeRoundRobin
-	DefaultAccessLog = "/dev/stdout"
+	// LbTypeRoundRobin is the name for roundrobin LB
 	LbTypeRoundRobin = "round_robin"
-	RDSName          = "rds"
 
 	// HTTPConnectionManager is the name of HTTP filter.
 	HTTPConnectionManager = "http_connection_manager"
+
 	// TCPProxyFilter is the name of the TCP Proxy network filter.
 	TCPProxyFilter = "tcp_proxy"
 
-	// URI HTTP header
+	// HeaderURI is URI HTTP header
 	HeaderURI = "uri"
 
 	// WildcardAddress binds to all IP addresses
 	WildcardAddress = "0.0.0.0"
 )
 
+var (
+	// DefaultMeshConfig configuration
+	DefaultMeshConfig = &proxyconfig.ProxyMeshConfig{
+		DiscoveryAddress: "manager:8080",
+
+		ProxyListenPort:        15001,
+		ProxyAdminPort:         15000,
+		DrainDuration:          ptypes.DurationProto(2 * time.Second),
+		ParentShutdownDuration: ptypes.DurationProto(3 * time.Second),
+		DiscoveryRefreshDelay:  ptypes.DurationProto(1 * time.Second),
+		ConnectTimeout:         ptypes.DurationProto(1 * time.Second),
+		IstioServiceCluster:    "istio-proxy",
+
+		AuthPolicy:    proxyconfig.ProxyMeshConfig_NONE,
+		AuthCertsPath: "/etc/certs",
+	}
+)
+
+// convertDuration converts to golang duration and logs errors
+func convertDuration(d *duration.Duration) time.Duration {
+	dur, err := ptypes.Duration(d)
+	if err != nil {
+		glog.Warningf("error converting duration %#v: %v", d, err)
```

Comment:
Early and completion validation sounds good. In that case, we should probably treat the warning here as an error/fatal/panic since it's clearly a programmer error at that point and otherwise envoy might be incorrectly configured?